macrosynergy/download/dataquery.py

```diff
@@ -285,540 +317,629 @@ def _get_token(self) -> str:
+        download_outputs: List[List[Dict]] = []
+        failed_batches: List[List[str]] = []
 
-        thread_output = []
-        final_output = []
-        error_tickers = []
-        error_messages = []
         if self.concurrent:
+            future_objects: List[concurrent.futures.Future] = []
 
             with concurrent.futures.ThreadPoolExecutor() as executor:
-                futures = []
-
-                for r_list in tqdm(
-                    tick_list_compr,
+                for ib, expr_batch in tqdm(
+                    enumerate(expr_batches),
+                    desc="Requesting data",
                     disable=not show_progress,
-                    desc="Requesting data : ",
+                    total=len(expr_batches),
                 ):
-
-                    params_copy = params.copy()
-                    params_copy["expressions"] = r_list
-                    futures.append(
-                        [
-                            executor.submit(
-                                self._fetch_threading, endpoint, params_copy
-                            ),
-                            r_list,
-                        ]
+                    curr_params: Dict = params_dict.copy()
+                    curr_params["expressions"] = expr_batch
+                    future_objects.append(
+                        executor.submit(
+                            self._request_thread,
+                            url=self.access_method.base_url + endpoint,
+                            params=curr_params,
+                            proxy=self.proxy,
+                            tracking_id=f"request_{ib}",
+                        )
                     )
-
-                    time.sleep(delay)
-                    thread_output.append(futures[-1][0])
-
-                for i, fto in tqdm(
-                    enumerate(concurrent.futures.as_completed(thread_output)),
+                for ib, future in tqdm(
+                    enumerate(future_objects),
+                    desc="Downloading data",
                     disable=not show_progress,
-                    desc="Downloading data : ",
-                    total=len(thread_output),
+                    total=len(future_objects),
                 ):
                     try:
-                        response, status, msg = fto.result()
-                        if not status:
-                            error_tickers.extend(tick_list_compr[i])
-                            error_messages.append(msg)
-                            logger.warning(
-                                f"Error in requestion tickers: {', '.join(futures[i][1])}."
-                                f"Error details: {msg}"
-                            )
-
-                        if fto.__dict__["_result"][0] is None:
-                            return None
-
+                        result = future.result()
+                        download_outputs.append(result)
```

Comment:
Why not do it in a single step:
```python
download_outputs.append(future.result())
```
Unless you only want to append if not None.

Reply:
Done